Navbar

indigorose opened the pull request:
Added a navbar

indigorose committed:
Update: navbar

indigorose committed:
Update: remove merge changes

webtechalex commented:
I think the linting might not be working properly on this one. We have some extra spaces and some places where it has switched the quote style. I'll have a look into it. Maybe we can get this change merged so that we have the feature at least, then I (or anyone else) can have a look at the code style stuff

indigorose commented:
That's fine, it might be clashes with the prettier files when configuring node last night. 

I forgot to mention yesterday that I also included some minor changes to the pages for responsivity on mobile/smaller screens.

webtechalex commented on src/components/Navbar.astro:
```diff
@@ -0,0 +1,104 @@
+</nav>
+
+<script is:inline>
+	const menuButton = document.getElementById('menu');
```
I wonder if there is a more idiomatic way to do this. Maybe it's just my React-centered brain that has made me a little allergic to the DOM API and this sort of thing is perfectly fine in astro 🤔 ...not sure. Any thoughts @cajogos ?

webtechalex commented on src/pages/coding-challenges.astro:
```diff
@@ -1,11 +1,12 @@
 ---
 import Layout from "../layouts/Layout.astro";
-import Logo from "../components/Logo.astro";
+// import Logo from "../components/Logo.astro";
```
I think this one wants deleting 😁

indigorose replied:
Happy to remove this.

webtechalex replied:
Awesome. I'll watch for the change.
If you have time, can you also just give the eslint and prettier scripts a kick and commit and push any changes they make?
```
npm run lint
npm run format
```
Maybe we can clean up some of these extra spaces and make the PR smaller.

webtechalex replied:
No rush on these by the way! I know it's getting late! 😬 😁

webtechalex commented on src/pages/blog.astro:
```diff
@@ -0,0 +1,12 @@
+
+<Layout title="Blog">
+<Navbar />
+
```
Nitpick: redundant whitespace

indigorose committed:
Edited navbar, whitespace and styling

indigorose commented:
Updated the quotations and whitespace where I can. Noted that I made a change the FAQ cards, they weren't responsive on smaller screens but my solution has a glitch and reverted this back to the original for now. This could be another first issue for others.

webtechalex commented:
We still have large amounts of unnecessary changes due to extra whitespace added in FAQCard.astro, and index.astro. ~@indigorose can you pull the spacing back in those files so that it matches the originals? I was hoping that if you ran the linting and formatting scripts that it would be done automatically, but maybe not? Once done and merged I will take a look at the eslint and prettier configs and try to figure out why the checks are passing when they should fail.~
Actually, scratch that. I'm going to merge it as it is and I can use the current state to debug the problem with the lint and format rules.

webtechalex commented:
On the subject of the responsiveness changes: I definitely think it's a good idea to keep PRs focused on a single issue or task. Awesome that you want to GSD but no need to cram things and risk having to context-switch 😉
